Core/src/org/sleuthkit/autopsy/modules/plaso/PlasoIngestModule.java

```diff
@@ -0,0 +1,410 @@
+                        logger.log(Level.INFO, Bundle.PlasoWorker_tempDir_deletion_execution_error());
+                    }
+                } catch (Exception ex) {
+                    logger.log(Level.INFO, Bundle.PlasoWorker_tempDir_deletion_execution_error(), ex);
+                }
+                IngestMessage message = IngestMessage.createMessage(IngestMessage.MessageType.DATA, Bundle.PlasoWorker_has_run(), Bundle.PlasoWorker_completed() );
+                IngestServices.getInstance().postMessage(message);
+
+            } catch (Exception ex) {
+                logger.log(Level.SEVERE, Bundle.PlasoWorker_exception_running_plaso(), ex);
+        }
+
+        }
+  
+        @NbBundle.Messages({
+            "PlasoWorker_exception_posting_artifact=Exception Posting artifact.",
+            "PlasoWorker_event_datetime=Event Date Time",
+            "PlasoWorker_event_description=Event Description",
+            "PlasoWorker_exception_adding_artifact=Exception Adding Artifact",
+            "PlasoWorker_exception_database_error=Error while trying to read into a sqlite db."
+        })
+        private void createPlasoArtifacts(String plasoDb) {
+            org.sleuthkit.datamodel.Blackboard blackboard;
+            SleuthkitCase sleuthkitCase = Case.getCurrentCase().getSleuthkitCase();
+            blackboard = sleuthkitCase.getBlackboard();
+            ResultSet resultSet;
+            String connectionString = "jdbc:sqlite:" + plasoDb; //NON-NLS
+            String sqlStatement = "select substr(filename,1) filename, strftime('%s', datetime) 'TSK_DATETIME', description 'TSK_DESCRIPTION' \n" +
+                                  "  from log2timeline where source != 'FILE';";
+//            String sqlStatement = "select datetime('unixepoch', datetime) 'TSK_DATETIME', description 'TSK_DESCRIPTION' "
+//                                +  " from log2timeline where source != 'REG';";
+            try {
+                SQLiteDBConnect tempdbconnect = new SQLiteDBConnect("org.sqlite.JDBC", connectionString); //NON-NLS
+                resultSet = tempdbconnect.executeQry(sqlStatement);
+                while (resultSet.next()) {
+                    Collection<BlackboardAttribute> bbattributes = new ArrayList<>();
+                    bbattributes.add(new BlackboardAttribute(ATTRIBUTE_TYPE.TSK_DATETIME, Bundle.PlasoWorker_event_datetime(), resultSet.getLong("TSK_DATETIME")));
+                    bbattributes.add(new BlackboardAttribute(ATTRIBUTE_TYPE.TSK_DESCRIPTION, Bundle.PlasoWorker_event_description(), resultSet.getString("TSK_DESCRIPTION")));
+                    
+                    List<AbstractFile> resolvedFiles = getAbstractFile(resultSet.getString("filename"));
+                    try {
+                        for (AbstractFile resolvedFile : resolvedFiles) {
+                            BlackboardArtifact bbart = resolvedFile.newArtifact(BlackboardArtifact.ARTIFACT_TYPE.TSK_EVENT);
```

Comment:
https://github.com/sleuthkit/sleuthkit/pull/1260  renames TSK_EVENT to TSK_TL_EVENT.  It also introduces two new artifact types you can use the same way, but for more specific events: TSK_LOG_EVENT and TSK_REGISTRY_EVENT.

For TSK_TL_EVENTs. If possible please use the TSK_EVENT_TYPE attribute to indicate the event type.  The value should be the id of one of the instances of EventType available as static members of the interface.  i.g. EventType.EMAIL.getTypeID().  Use EventType.OTHER.getTypeID() if none of the available ones fit.